test/e2e/nodemaintenance_test.go

```diff
@@ -354,3 +362,74 @@ func checkFailureStatus(t *testing.T, f *framework.Framework) {
 		}
 	}
 }
+
+func getOperatorHostname(t *testing.T, f *framework.Framework) string {
+	pods := &corev1.PodList{}
+
+	err := f.Client.List(goctx.TODO(), &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(operatorLabel),
+	}, pods)
+
+	if err != nil {
+		t.Fatalf("failed listing operator pods: %v", err)
+	}
+
+	if len(pods.Items) < 1 {
+		t.Fatal("there are no operator pods")
+	}
+
+	return pods.Items[0].Spec.NodeName
+}
+
+func countWorkerNodes(t *testing.T, f *framework.Framework) int {
+	nodes := corev1.NodeList{}
```

Comment:
another suggestion to get all nodes with a label:
```
// GetByRole returns all nodes with the specified role
func GetByRole(c client.Client, role string) ([]corev1.Node, error) {
	selector, err := labels.Parse(fmt.Sprintf("%s/%s=", testutils.LabelRole, role))
	if err != nil {
		return nil, err
	}

	nodes := &corev1.NodeList{}
	if err := c.List(context.TODO(), nodes, &client.ListOptions{LabelSelector: selector}); err != nil {
		return nil, err
	}
	return nodes.Items, nil
}
```

Can move to a util class

Reply:
Problem is, I don't think worker nodes have roles applied (or not all of the time):
```
$ kubectl get nodes
NAME     STATUS   ROLES    AGE   VERSION
node01   Ready    master   20h   v1.16.2
node02   Ready    <none>   20h   v1.16.2
node03   Ready    <none>   20h   v1.16.2
```

Reply:
right.  well if some of the code lines here are more elegant feel free to use them anyway